apps/condo/domains/marketplace/components/MarketplacePageContent/index.tsx

```diff
@@ -51,8 +57,37 @@ export const MarketplacePageContent = () => {
     })
 
     const marketplaceIsSetup = invoiceContext && get(invoiceContext, 'status') === INVOICE_CONTEXT_STATUS_FINISHED
-    const canReadPayments = get(userOrganization, ['link', 'role', 'canReadPayments'], false)
-    const canManageInvoices = get(userOrganization, ['link', 'role', 'canManageInvoices'], false)
+    const role = get(userOrganization, ['link', 'role'], {})
+    const canReadPayments = get(role, ['canReadPayments'], false)
+    const canManageInvoices = get(role, ['canManageInvoices'], false)
+
+    const filterMetas = useFilters(currentTab)
+    const { filtersToWhere, sortersToSortBy } = useQueryMappers(filterMetas, [])
+    const { filters, sorters } = parseQuery(router.query)
+
+    const [isStatusDescModalVisible, setIsStatusDescModalVisible] = useState<boolean>(false)
+    const [titleStatusDescModal, setTitleStatusDescModal] = useState('')
+    const [textStatusDescModal, setTextStatusDescModal] = useState('')
+    const openStatusDescModal = (statusType) => {
+        const titleModal = intl.formatMessage({ id: 'payment.status.description.title.' + statusType })
+        const textModal = intl.formatMessage({ id: 'payment.status.description.text.' + statusType })
+
+        setTitleStatusDescModal(titleModal)
+        setTextStatusDescModal(textModal)
+        setIsStatusDescModalVisible(true)
+    }
+    const tableColumns = useTableColumns(filterMetas, currentTab, openStatusDescModal)
+
+    const searchPaymentsQuery = useMemo(() => {
+        return {
+            invoice: {
+                context: {
+                    organization: { id: orgId },
+                },
+            },
+        }},
+    [filters, filtersToWhere, orgId])
+    const sortBy = useMemo(() => sortersToSortBy(sorters) as SortPaymentsBy[], [sorters, sortersToSortBy])
```

Comment:
This all related to `MarketplacePaymentsContent`. Put it inside the `MarketplacePaymentsContent` component pls